ui/components/component-library/button-base/button-base.test.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable jest/require-top-level-describe */
+import { render } from '@testing-library/react';
+import React from 'react';
+import { BUTTON_SIZES } from './button.constants';
+import { ButtonBase } from './button-base';
+
+describe('ButtonBase', () => {
+  it('should render button element correctly', () => {
+    const { getByTestId, container } = render(
+      <ButtonBase data-testid="button-base" />,
+    );
+    expect(getByTestId('button-base')).toBeDefined();
+    expect(container.querySelector('button')).toBeDefined();
+    expect(getByTestId('button-base')).toHaveClass('mm-button');
+  });
+
+  it('should render anchor element correctly', () => {
+    const { getByTestId, container } = render(
+      <ButtonBase as="a" data-testid="button-base" />,
+    );
+    expect(getByTestId('button-base')).toBeDefined();
+    expect(container.querySelector('button')).toBeDefined();
```

Comment:
I'm confused how this still passing? It's looking for a button element but it should be rendering an `<a>` tag

```suggestion
    expect(container.querySelector('a')).toBeDefined();
````

Reply:
Weird. Oh well it's working haha